app/controllers/videos_controller.rb

```diff
@@ -15,4 +15,15 @@ def search
     @videos = Video.search_by_title(params[:search_string])
   end
 
+  def advanced_search
+    options = { reviews: params[:reviews] }
+    options[:rating_from] = params[:rating_from] if params[:rating_from].present?
+    options[:rating_to] = params[:rating_to] if params[:rating_to].present?
+
```

Comment:
You could also move the check to the search. That way it would work no matter where the search was initiated from. Currently the only place with the verification is here.